client.go

```diff
@@ -72,13 +81,15 @@ type client struct {
 	seedBrokers []*Broker
 	deadSeeds   []*Broker
 
-	brokers  map[int32]*Broker                       // maps broker ids to brokers
-	metadata map[string]map[int32]*PartitionMetadata // maps topics to partition ids to metadata
+	brokers      map[int32]*Broker                       // maps broker ids to brokers
+	metadata     map[string]map[int32]*PartitionMetadata // maps topics to partition ids to metadata
+	coordinators map[string]*Broker                      // Maps consumer group names to coordinating brokers
```

Comment:
The tradeoff is the subtleties of updating everything. AFAICT the code is currently fine, but it opens us to cases where a broker ends up in the coordinator map but not in the broker map at all, which is then a leak on shutdown and could lead to weird behaviour. Making `brokers` the source of truth and have everything point into that seems safer.